storage/statedb/database_test.go

```diff
@@ -86,7 +86,8 @@ func TestDatabase_DeReference(t *testing.T) {
 
 func TestDatabase_Size(t *testing.T) {
```

Comment:
I think you need to test `getTrieNodeCacheSizeMB()` with different `common.TotalPhysicalMemGB`.